cuML/DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,114 @@
+# cuML developer guide
```

Comment:
For completeness, it seems like DEVELOPER_GUIDE.md and [CONTRIBUTING.md](https://github.com/rapidsai/cuml/blob/master/CONTRIBUTING.md) should cross reference each other somehow.

Reply:
Make sense. Just added a cross reference.